### PR TITLE
Add docs for lib bindings with supported library versions

### PR DESCRIPTION
**Pull request by straight-shoota**
Adds documentation about which library versions are supported by the various lib bindings.

I'm planning to put that information on https://crystal-lang.org/reference/1.13/man/required_libraries.html as well. The URLs to this page should hopefully remind us to sync updates there.

Ref https://github.com/crystal-lang/crystal/issues/11358#issuecomment-2268395035

### CODE REVIEW

**Comment by Blacksmoke16** [comment]
Would be nice if we could leverage/improve https://github.com/crystal-lang/shards/blob/655a3b0fc416986dae1846ded3856aaf46198085/docs/shard.yml.adoc?plain=1#L176-L205 more to have this available in a more idiomatic way.

File: src/llvm/lib_llvm.cr [hunk 1]
```diff
@@ -40,6 +40,11 @@
   end
 {% end %}
 
+# Supported library versions:
+#
+# * LLVM (8-18; aarch64 requires 12+)
```
Comment (HertzDevil) [inline src/llvm/lib_llvm.cr:45]:
```suggestion
# * LLVM (8-18; aarch64 requires 13+)
```

**Comment by straight-shoota** [comment]
@Blacksmoke16 Yes, a structured format could be nice. But not sure how that would look like. A plain mapping between lib name and a version selector is not always sufficient.
So it's best to start with this and maybe figure out some more elaborate notation later.

File: src/crystal/lib_iconv.cr [hunk 2]
```diff
@@ -4,6 +4,11 @@ require "c/stddef"
   {% raise "The `without_iconv` flag is preventing you to use the LibIconv module" %}
 {% end %}
 
+# Supported library versions:
+#
+# * libiconv-gnu
```
Comment (ysbaddaden) [inline src/crystal/lib_iconv.cr:9]: Don't we use any iconv library provided by the system? It's a POSIX standard.

Comment (straight-shoota) [reply src/crystal/lib_iconv.cr:9]: Naming is a bit confusing unfortunately. But AFAIU the POSIX standard is just called `iconv` without `lib`. We have bindings for that as part of `LibC`.
`libiconv` as an explicit library is only used on Windows and Android (< API 28), or when explicitly asked with `-Duse_libiconv`. And then we only use the GNU variant of the library.

https://github.com/crystal-lang/crystal/blob/401eb47bf373f11b2da6e382c306ead36615e1d6/src/crystal/iconv.cr#L1-L7

**Review by ysbaddaden — APPROVED** [review]

**Pull request merged**